numba_cuda/numba/cuda/cudadrv/driver.py

```diff
@@ -2916,6 +2921,184 @@ def complete(self):
         """
 
 
+class _Linker(_LinkerBase):
```

Comment:
It seems like this class is the interface used elsewhere - e.g. in `codegen`, so implying it is private seems a little strange. Should it be
```suggestion
class Linker(_LinkerBase):
```

?

Reply:
Since we're pushing `cuda.core.Linker` as the public facing linker interface, I tried to avoid having something else called that in the codebase. We don't have to make it private though, it only is due to https://github.com/NVIDIA/numba-cuda/pull/133#discussion_r2130488367.